src/dashboard/modals/whatup-modals/modal.tsx

```diff
@@ -79,97 +76,75 @@ const Modal: FC<{ orderId: string }> = (props) => {
+                                    </SectionHelper>
+                                )}
+                                <FormField label={"Select an item:"}>
+                                    <RadioGroup
+                                        value={selectedProduct.id || ''}
+                                        key={selectedProduct.id}
+                                        onChange={(prodId) => {
+                                            const product = productOptionsList.find(p => {
+                                                return p.id === prodId;
+                                            });
+                                            setSelectedProduct(product!);
+                                        }
+                                        }
+                                    >
+                                        {productOptionsList.map((product) => {
+                                            return (<RadioGroup.Radio value={product.id}>
+                                                <Box gap={2} verticalAlign="middle">
+                                                    <Image
+                                                        width={'45px'}
+                                                        height={'45px'}
+                                                        src={getImageFromWixMedia(product.image)}/>
+                                                    <Box direction="vertical">
+                                                        <Text weight="normal">{product.value}</Text>
+                                                    </Box>
+                                                </Box>
+                                            </RadioGroup.Radio>)
+                                        })}
+                                    </RadioGroup>
+                                </FormField>
+                                <Divider skin="light"/>
+                                <FormField label="WhatsApp message to the user">
+                                    <InputArea
+                                        minHeight={'270px'}
+                                        placeholder=""
+                                        rows={6}
+                                        maxLength={500}
+                                        hasCounter
+                                        resizable
+                                        value={controlledMessageForInputArea}
+                                        onChange={(e) => setControlledWhatsappMessage(e.target.value)}
+                                    />
+                                </FormField>
+                            </>
+                        )}
                     </Box>
                 }
             />
         </WixDesignSystemProvider>
     );
 };
 
-
-const renderOrder = async (orderId: string) => {
```

Comment:
Isnt it better to keep this part of code here instead of inside the useEffect?

https://github.com/RoyNoymanW/order-custom-action/pull/1/files#diff-1535971ff90563eb125c2cbbcc8755fe20c8b0afad6f20eed9111eaca94dbf9cL52-L58

Reply:
I don't see why, we usually put it there